compiler/rustc_session/src/session.rs

```diff
@@ -1553,11 +1570,18 @@ fn early_error_handler(output: config::ErrorOutputType) -> rustc_errors::Handler
                 false,
                 None,
                 false,
+                false,
             ))
         }
-        config::ErrorOutputType::Json { pretty, json_rendered } => {
-            Box::new(JsonEmitter::basic(pretty, json_rendered, None, fallback_bundle, None, false))
-        }
+        config::ErrorOutputType::Json { pretty, json_rendered } => Box::new(JsonEmitter::basic(
+            pretty,
+            json_rendered,
+            None,
+            fallback_bundle,
+            None,
+            false,
+            false,
```

Comment:
I think if you don't set this for JSON output, it won't show up when using cargo.

Reply:
I'm not sure what you mean with that. Afaik most code that calls this function doesn't have the sessions options available to it (because it's creating it). So i'm not sure what to do here.

Reply:
Oh, this is in `early_error`, I misread. Ok, seems fine to skip for now. I think it shouldn't be too hard to add it though, nearly every place that calls this has access to `Options` and those that don't can still call `matches.opt_get("track-diagnostics")` before calling the error code.